### DISCUSSION TITLE
Min SoC Set to 99%!

### DISCUSSION THREAD

**Discussion by C00K50N** [nathanmarlor/foxess_em#31] [Q&A] [answered]
Not sure if I'm the only one, but I woke up this morning to a full battery, having not discharged any in the couple hours after the cheap period ended.  I checked my Fox Cloud app and the Min SoC and Min SoC (On Grid) set were both set at 99%...

![image](https://user-images.githubusercontent.com/102669305/197380592-4bf5ef62-5318-46cf-9ba0-b4bcdaef398f.png)

I checked the battery config options in this integration and it's still set to 10%...

![image](https://user-images.githubusercontent.com/102669305/197380762-1c4cc9a4-02d0-4156-ae05-5e574e537a53.png)

@nathanmarlor do you change this value for the duration of the cheap period in order to hold after a charge?  If so, perhaps it didn't change back to 10% at the end of the cheap period?  Something you could possibly look into at some point?

**Comment by nathanmarlor** [comment, marked as answer]
I woke up to the same...

Spotted something daft in the code which makes it crash like that on the second day, fixed and pushed out v1.1.1 this morning :)

For reference - the charge service stores the force charge cancellation callback (used to remove the SoC watcher once we've hit the target SoC) in a list. As part of clean up the list is iterated and all the callbacks invoked. This works fine on the first day, however as the list was never cleared on the second day it'll also try to invoke the callback from the first day which caused an exception; this prevented the rest of the method from running which reset the min SoC etc. The list is a hangover from the previous implementation - will tidy up a little more when I have time, but fixed for now.

**Reply by C00K50N** [reply]
Great, thank you.  I wondered if the update this morning was to fix it, but I'm afraid I wasn't able to understand the terms.  I'll update today.
